### PR TITLE
make foreman puppet module dir optional

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
This is currently work in progress. I'm not sure yet if the tests pass and I also need to adjust the docs. @ekohl @mmoll feedback is highly appreciated :)

----

I noticed that the module creates this directory. I never saw a foreman installation that uses this so I want to make it optional (or get rid of the exec at all). If there are usecases for the directory I will change the default from false to true.

File: manifests/server/config.pp [hunk 1]
```diff
@@ -125,9 +125,11 @@
   }
 
   # Create Foreman share dir which does not depend on Puppet version
-  exec { 'mkdir -p /usr/share/puppet/modules':
-    creates => '/usr/share/puppet/modules',
-    path    => ['/usr/bin', '/bin'],
+  if ($puppet::server::create_foreman_modules_directory == true) {
```
Comment (ekohl) [inline manifests/server/config.pp]: I think we can avoid a parameter:
```puppet
if '/usr/share/puppet/modules' in $puppet::server::common_modules_path {}
```

**Comment by bastelfreak** [comment]
@ekohl I updated the code

File: manifests/server/config.pp [hunk 2]
```diff
@@ -124,10 +124,12 @@
     mode   => '0750',
   }
 
-  # Create Foreman share dir which does not depend on Puppet version
-  exec { 'mkdir -p /usr/share/puppet/modules':
-    creates => '/usr/share/puppet/modules',
-    path    => ['/usr/bin', '/bin'],
+  if ('/usr/share/puppet/modules' in $puppet::server::common_modules_path) {
```
Comment (ekohl) [inline manifests/server/config.pp]: the parenthesis aren't needed

**Comment by mmoll** [comment]
merged, danke @bastelfreak!

**Pull request merged**